server/src/resolvers/Mutation.js

```diff
@@ -0,0 +1,58 @@
+module.exports.Mutation = {
+  signInWithGoogle: async (_, {googleId}, {dataSources}) => {
+    const user = await dataSources.mainAPI.findOneUser({googleId});
+    if (user === null) {
+      return {
+        success: false,
+        message: 'User not found. You have to sign up first.',
+        token: null,
+        user: null,
+      };
+    }
+
+    // TODO(lsh9034): Implement session logic.
+    //  Create session and return sessionId to client.
+
+    return {
+      success: true,
+      message: 'Success',
+      token: 'We have to implement this',
+      user,
+    };
+  },
+
+  signUpWithGoogle:
+      async (
+          _, {googleId, email, firstName, lastName, profileImgUrl},
+          {dataSources}) => {
+        const user = await dataSources.mainAPI.findOrCreateUser({googleId},
+            {
+              email,
+              firstName,
+              lastName,
+              profileImgUrl: profileImgUrl ? profileImgUrl : '',
+            });
+
+        if (user === null) {
+          return {
+            success: false,
+            message: 'Fail to create the user. Please try again later',
+            token: null,
+            user: null,
+          };
+        }
+
+        // TODO(lsh9034): Implement session logic.
+        //  Create session and return sessionId to client.
```

Comment:
Why two space after //

Reply:
The second line is about TODO. WebStorm(IDE) highlighted those two lines as TODO. But if I remove a space after // on the second line. The second line is not highlighted.
Should I remove spaces?
If TODO comment is over many lines, what is the convention?

Reply:
From the look of it, we can replace the first line with second line. 

In general, TODO msgs are short (don't need to describe the pseudo code there). Let's only use 1 space after // overall.

Reply:
Ok. I will modify all of them. Thank you!